scala/private/rule_impls.bzl

```diff
@@ -1006,36 +1013,6 @@ def scala_test_impl(ctx):
         jars.jars2labels,
     )
 
-    # _scalatest is an http_jar, so its compile jar is run through ijar
```

Comment:
why is this all removed now? Is this because we shouldn't depend on a test anyway?

Reply:
Good question. 
I have an answer which I’m fairly confident about but unfortunately  not 100%.
The answer is that since I added the scalatest label as part of the base classpath it’s jars and labels are collected in the main flow.
I have to say that no tests failed even before I did this so I think this area of scalatest and integration with strict-deps/unused deps isn’t tested well.

Lastly tests can depend on tests, not a problem also in java.